internal/utils/main.go

```diff
@@ -1101,3 +1104,45 @@ type HistoryArchiveLedgerAndLCM struct {
 	Ledger historyarchive.Ledger
 	LCM    xdr.LedgerCloseMeta
 }
+
+type ChangeDetails struct {
+	ClosedAt       time.Time
+	LedgerSequence uint32
+	OperationType  null.Int
+	TransactionID  null.Int
+	OperationID    null.Int
+}
+
+func GetChangesDetails(ledgerChange ingest.Change) ChangeDetails {
+	var outputOperationType null.Int
+	var outputTransactionID null.Int
+	var outputOperationID null.Int
+	var lcm *xdr.LedgerCloseMeta
+
+	if ledgerChange.Ledger == nil {
+		lcm = &ledgerChange.Transaction.Ledger
+	} else {
+		lcm = ledgerChange.Ledger
+	}
+	closedAt := ledger.ClosedAt(*lcm)
+	ledgerSequence := ledger.Sequence(*lcm)
+	if ledgerChange.Transaction != nil {
+		outputTransactionID = null.NewInt(toid.New(int32(ledgerSequence), int32(ledgerChange.Transaction.Index), 0).ToInt64(), true)
+		// outputOperationID may point to the incorrect operation because the change compactor takes the latest OperationIndex if it exists
+		// It is possible that OperationIndex is 0 and invalid but will be used to populate outputOperationID anyways
+		// TODO: This should be fixed when the LedgerEntryChange transforms/processors use the actual changes instead of compacted changes
+		outputOperationID = null.NewInt(toid.New(int32(ledgerSequence), int32(ledgerChange.Transaction.Index), int32(ledgerChange.OperationIndex+1)).ToInt64(), true)
+		operation, ok := ledgerChange.Transaction.GetOperation(ledgerChange.OperationIndex)
```

Comment:
Just confirming: there is no way that the OperationType will get out of sync with the OperationIndex? It's not possible to cross them and have a mismatched OperationID <> OperationType, correct?